### PR TITLE
Skip rows for faster scan of data.

### PR DESCRIPTION
**Pull request by bsura**

### CODE REVIEW

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java [hunk 1]
```diff
@@ -382,13 +382,121 @@ public Object call(ArrayList<ArrayList<KeyValue>> rows) throws Exception {
...
+        String metric = _convertToRegex(query.getMetric());
+        String tagKey = _convertToRegex(query.getTagKey());
+        String tagValue = _convertToRegex(query.getTagValue());
+        String rowKeyRegex = "^" + _constructRowKey(namespace, scope, metric, tagKey, tagValue, metadata.type) + "$";
+    	
+    	List<ScanFilter> filters = new ArrayList<ScanFilter>();
+    	filters.add(new RowFilter(CompareOp.EQUAL, new RegexStringComparator(rowKeyRegex)));
+        filters.add(new KeyOnlyFilter());
+        filters.add(new FirstKeyOnlyFilter());
+        FilterList fl = new FilterList(filters, FilterList.Operator.MUST_PASS_ALL);
```
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java]: Change var name to filterList

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java [hunk 2]
```diff
@@ -382,13 +382,121 @@ public Object call(ArrayList<ArrayList<KeyValue>> rows) throws Exception {
...
+    
+    private List<String> _getUniqueFastScan(MetricSchemaRecordQuery query, final int limit, final RecordType type) {
+    	requireNotDisposed();
+    	SystemAssert.requireArgument(RecordType.METRIC.equals(type) || RecordType.SCOPE.equals(type), 
+    			"This method is only for use with metric or scope.");
+    	
+    	final List<String> result = new ArrayList<>();
+    	
+    	final ScanMetadata metadata = _constructScanMetadata(query);
+        String namespace = _convertToRegex(query.getNamespace());
```
Comment (bsura) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java]: We are not encoding anything. We are just converting the expression from wildcards to use java regex. A dot(.) which has no special meaning in wildcarding would become (\\.). Similarly * wildcard would become (.*).

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java [hunk 3]
```diff
@@ -382,13 +382,121 @@ public Object call(ArrayList<ArrayList<KeyValue>> rows) throws Exception {
...
+			}
+		});
+        
+        try {
+			ArrayList<ArrayList<KeyValue>> result = deferred.join(TIMEOUT);
+			return result;
+		} catch (InterruptedException e) {
+			throw new SystemException("Interrupted while waiting to obtain results for query", e);
+		} catch (TimeoutException e) {
+			_logger.warn("Timed out while waiting to obtain results.");
```
Comment (bsura) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java]: Hmm. Need to give more thought.

**Comment by bsura** [comment]
@dilipdevaraj-sfdc Can we merge this now?

**Pull request merged**